.NET/Microsoft.Recognizers.Text.NumberWithUnit/Utilities/DictionaryUtils.cs

```diff
@@ -31,12 +32,21 @@ public static void BindUnitsString(IDictionary<string, string> sourceDictionary,
 
             foreach (var token in values)
             {
-                if (string.IsNullOrWhiteSpace(token) || sourceDictionary.ContainsKey(token))
+                if (string.IsNullOrWhiteSpace(token) || (sourceDictionary.ContainsKey(token) && sourceDictionary[token].Equals(key)))
                 {
                     continue;
                 }
 
-                sourceDictionary.Add(token, key);
+                // This segment of code is used to prevent duplicated key-values in resource files.
+                try
+                {
+                    sourceDictionary.Add(token, key);
+                }
+                catch (ArgumentException exception)
+                {
+                    // Temporarily catch exception to show where errors occur, should be removed afterward.
```

Comment:
Shouldn't we remove this now, so the build breaks on tests if there are duplicate keys?

Reply:
Yes we should.